### PR TITLE
Added upgrade suite for 4.x to 5.x with RGW tests

### PR DESCRIPTION
**Pull request by vivekanandan-k-rh**
Added upgrade suite for 4.x to 5.x with RGW tests

# Test steps:
# ---------------------------------------------------------------------
# - Deploy Nautilus Ceph cluster using CDN RPMS with lvm osd scenario
# - Run RGW tests
# - Convert the cluster to containerized using ceph-ansible
# - Upgrade to Pacific using ceph-ansible
# - Run RGW tests after upgrade

Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ON34F0/

The following ticket is created to triage test failures: https://issues.redhat.com/browse/RHCEPHQE-1950

### CODE REVIEW

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+        grafana_admin_password: p@ssw0rd
+        node_exporter_container_image: registry.redhat.io/openshift4/ose-prometheus-node-exporter:v4.6
+        grafana_container_image: registry.redhat.io/rhceph/rhceph-4-dashboard-rhel8:4
+        prometheus_container_image: registry.redhat.io/openshift4/ose-prometheus:v4.6
+        alertmanager_container_image: registry.redhat.io/openshift4/ose-prometheus-alertmanager:v4.6
+        cephfs_pools:
+          - name: "cephfs_data"
+            pgs: "16"
+          - name: "cephfs_metadata"
+            pgs: "16"
```
Comment (udaysk23) [inline suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:52]: @vivekanandan-k-rh could you please look into this

Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml]: Removed those lines (cephfs_pools) as it is not required. Thanks @udaysk23 for the pointer!

**Review by udaysk23 — APPROVED** [review]
LGTM

File: pipeline/scripts/5/1/tier-1/test-upgrade-4x-to-5x-rgw-tests.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#! /bin/sh
+echo "Beginning RHCS 4x to 5x upgrade testing with RGW tests."
```
Comment (psathyan) [inline pipeline/scripts/5/1/tier-1/test-upgrade-4x-to-5x-rgw-tests.sh:2]: Can we move this test to a different folder? Preferably to `tier-3` folder. All the stages are executed against every build. The reason behind the request is that we already have two many tests in this folder ... moving it out would improve the feedback loop.

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+    module: test_ansible.py
+    config:
+      use_cdn: True
+      build: '4.x'
+      ansi_config:
+        ceph_origin: repository
+        ceph_repository: rhcs
+        ceph_repository_type: cdn
+        ceph_rhcs_version: 4
+        ceph_stable_release: nautilus
```
Comment (psathyan) [inline suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:31]: This has no impact as the `ceph_repository` is set to `rhcs`

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+        alertmanager_container_image: registry.redhat.io/openshift4/ose-prometheus-alertmanager:v4.6
+    desc: deploy ceph containerized 4.x cdn setup using ceph-ansible
+    destroy-cluster: False
+    abort-on-fail: true
+
+- test:
+    name: check-ceph-health
+    module: exec.py
+    config:
+      cmd: ceph -s
```
Comment (psathyan) [inline suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:56]: Please include `ceph versions` also.

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+# - Convert the cluster to containerized using ceph-ansible
+# - Upgrade to Pacific using ceph-ansible
+# - Run RGW tests after upgrade
+#
+#############################################################################
+tests:
+- test:
+    name: install ceph pre-requisites
+    module: install_prereq.py
+    abort-on-fail: True
```
Comment (psathyan) [inline suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:17]: Please switch to using `true | false`... i.e. bool is all small cases.

File: pipeline/scripts/5/1/tier-1/test-upgrade-4x-to-5x-rgw-tests.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    CLEANUP_ARGS="$CLEANUP_ARGS --cloud ibmc"
+fi
+
+$WORKSPACE/.venv/bin/python run.py --cleanup $instance_name $CLEANUP_ARGS
+
+if [ $? -ne 0 ]; then
+    echo "cleanup instance failed for instance $instance_name"
+fi
+
+exit ${return_code}
```
Comment (psathyan) [inline pipeline/scripts/5/1/tier-1/test-upgrade-4x-to-5x-rgw-tests.sh:48]: The total runtime of the test is around 3 hours. Can we avoid running this test in the nightly pipeline (against all builds)? We can include this in the long running (cron pipeline) one... we expect it to be up shortly

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml [hunk 7]
```diff
@@ -0,0 +1,412 @@
+#############################################################################
```
Comment (psathyan) [inline suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:1]: Could we rename the file to `tier-1_rgw_test-upgrade-4-to-latest.yaml`?

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml [hunk 8]
```diff
@@ -0,0 +1,412 @@
...
+
+- test:
+    name: Upgrade containerized ceph to 5.x latest
+    polarion-id: CEPH-83573679
+    module: test_ansible_upgrade.py
+    config:
+      build: '5.x'
+      ansi_config:
+        ceph_origin: distro
+        ceph_stable_release: pacific
```
Comment (psathyan) [inline suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:233]: we could avoid this setting

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml [hunk 9]
```diff
@@ -0,0 +1,412 @@
...
+        upgrade_ceph_packages: True
+        dashboard_admin_user: admin
+        dashboard_admin_password: p@ssw0rd
+        grafana_admin_user: admin
+        grafana_admin_password: p@ssw0rd
+        node_exporter_container_image: registry.redhat.io/openshift4/ose-prometheus-node-exporter:v4.6
+        grafana_container_image: registry.redhat.io/rhceph/rhceph-5-dashboard-rhel8:5
+        prometheus_container_image: registry.redhat.io/openshift4/ose-prometheus:v4.6
+        alertmanager_container_image: registry.redhat.io/openshift4/ose-prometheus-alertmanager:v4.6
+    desc: Test Ceph-Ansible rolling update 4.x cdn -> 5.x latest -> cephadm adopt
```
Comment (psathyan) [inline suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:253]: Could you fix the description?

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml [hunk 10]
```diff
@@ -0,0 +1,412 @@
...
+    abort-on-fail: True
+
+# Basic Bucket Operation Tests
```
Comment (psathyan) [inline suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:256]: After the update... let's look at the cluster health and ceph versions.

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml [hunk 11]
```diff
@@ -0,0 +1,412 @@
...
+    name: Resharding tests
+    desc: Reshading test - dynamic
+    polarion-id: CEPH-83571740
+    module: sanity_rgw.py
+    config:
+      script-name: test_dynamic_bucket_resharding.py
+      config-file-name: test_dynamic_resharding.yaml
+      timeout: 500
+
+
```
Comment (psathyan) [inline suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:412]: Please remove the blank lines.

File: conf/pacific/rgw/upgrade_small-cluster.yaml [hunk 12]
```diff
@@ -0,0 +1,38 @@
+globals:
```
[the remark this answers is not in the document]
Comment (vivekanandan-k-rh) [reply conf/pacific/rgw/upgrade_small-cluster.yaml:1]: Done

File: pipeline/scripts/5/1/tier-1/test-upgrade-4x-to-5x-rgw-tests.sh [hunk 13]
```diff
@@ -0,0 +1,48 @@
+#! /bin/sh
+echo "Beginning RHCS 4x to 5x upgrade testing with RGW tests."
+
+random_string=$(cat /dev/urandom | tr -cd 'a-z0-9' | head -c 5)
+instance_name="ci-${random_string}"
+platform="rhel-8"
+rhbuild="5.1"
+test_suite="suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml"
+test_conf="conf/pacific/rgw/upgrade_small-cluster.yaml"
```
[the remark this answers is not in the document]
Comment (vivekanandan-k-rh) [reply pipeline/scripts/5/1/tier-1/test-upgrade-4x-to-5x-rgw-tests.sh:9]: Done

File: pipeline/scripts/5/1/tier-1/test-upgrade-4x-to-5x-rgw-tests.sh (hunk 2 above)
Comment (vivekanandan-k-rh) [reply pipeline/scripts/5/1/tier-1/test-upgrade-4x-to-5x-rgw-tests.sh:2]: Done

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml (hunk 4 above)
Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:56]: Done

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml (hunk 3 above)
Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml:31]: Done. Keeping only  `ceph_repository: rhcs`

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml (hunk 7 above)
Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml]: Done. Renamed it to tier-3_rgw_test-upgrade-4-to-latest.yaml

File: pipeline/scripts/5/1/tier-1/test-upgrade-4x-to-5x-rgw-tests.sh (hunk 6 above)
Comment (vivekanandan-k-rh) [reply pipeline/scripts/5/1/tier-1/test-upgrade-4x-to-5x-rgw-tests.sh]: Removed sh file

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml (hunk 8 above)
Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml]: Removed "build: '5.x' " . I believe we need 'ceph_origin: distro' as it needs to upgrade it to the latest build which is nonGAed

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml (hunk 10 above)
Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml]: Done

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml (hunk 9 above)
Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml]: Done

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml (hunk 11 above)
Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml]: Done

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml (hunk 7 above)
Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml]: Renamed it to tier-3_rgw_test-upgrade-4-to-latest.yaml

File: suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml (hunk 5 above)
Comment (vivekanandan-k-rh) [reply suites/pacific/rgw/tier-1-upgrade_4x_to_5x_latest.yaml]: Done

**Review by psathyan — APPROVED** [review]

**Pull request merged**